src/htmx.js

```diff
@@ -1235,14 +1235,27 @@ return (function () {
 
         var INPUT_SELECTOR = 'input, textarea, select';
 
+        /** @type {Object<string, import("./htmx").HtmxTriggerSpecification[]>} */
+        let triggerSpecsCache = {}
+
         /**
          * @param {HTMLElement} elt
          * @returns {import("./htmx").HtmxTriggerSpecification[]}
          */
         function getTriggerSpecs(elt) {
             var explicitTrigger = getAttributeValue(elt, 'hx-trigger');
             var triggerSpecs = [];
-            if (explicitTrigger) {
+            var shouldEvaluateTrigger = !!explicitTrigger
+            // If an explicit trigger attribute is set, check if it has already been computed & cached
+            if (shouldEvaluateTrigger) {
+                var cachedTriggerSpecs = triggerSpecsCache[explicitTrigger]
+                if (cachedTriggerSpecs) {
+                    triggerSpecs = cachedTriggerSpecs
+                    // If it had already been cached, we don't need to evaluate the trigger expression anymore
+                    shouldEvaluateTrigger = false
+                }
+            }
+            if (shouldEvaluateTrigger) {
```

Comment:
Can you pull this into a separate function called `parseTrigger`? That will let you write lines 1250-1258 as:

```js
if (shouldEvaluateTrigger) {
  var triggerSpecs = triggerSpecsCache[explicitTrigger] || parseTrigger(explicitTrigger)
}
```

which is smaller and I think communicates the intent better than reassigning the `if` condition